lib/Service/OpenAiAPIService.php

```diff
@@ -111,10 +111,15 @@ public function getModels(string $userId): array {
 			$this->areCredsValid = false;
 			throw $e;
 		}
-		if (!isset($modelsResponse['data'])) {
+		if (isset($modelsResponse['error'])) {
 			$this->logger->warning('Error retrieving models: ' . \json_encode($modelsResponse));
 			$this->areCredsValid = false;
-			throw new Exception($this->l10n->t('Unknown models error'), Http::STATUS_INTERNAL_SERVER_ERROR);
+			throw new Exception($modelsResponse['error'], Http::STATUS_INTERNAL_SERVER_ERROR);
+		}
+
+		if (!isset($modelsResponse['data'])) {
```

Comment:
Isn't it a bit optimistic to say
* the request didn't fail
* there is no data prop
* so the payload is the model list

?

In this case, I think we should try to check a little bit more if the response is valid. We could check if it's a list, take the first item and check it's valid. Wdyt?

Reply:
yeah it kind of depended on the api to report errors properly
sounds good